sympy/polys/fields.py

```diff
@@ -214,16 +218,24 @@ def _rebuild(expr):
                 return reduce(add, list(map(_rebuild, expr.args)))
             elif expr.is_Mul:
                 return reduce(mul, list(map(_rebuild, expr.args)))
-            elif expr.is_Pow and expr.exp.is_Integer:
-                return _rebuild(expr.base)**int(expr.exp)
-            else:
-                try:
-                    return domain.convert(expr)
-                except CoercionFailed:
-                    if not domain.is_Field and domain.has_assoc_Field:
-                        return domain.get_field().convert(expr)
-                    else:
-                        raise
+            elif expr.is_Pow or isinstance(expr, ExpBase):
+                b, e = expr.as_base_exp()
+                # look for bg**eg whose integer power may be b**e
+                choices = tuple((gen, bg, eg) for gen, bg, eg in powers
```

Comment:
This could have `for gen, (bg, eg) in powers` if `*` is omitted from `*gen.as_base_exp()` above.